src/main/java/com/arpnetworking/clusteraggregator/ClusterStatusCache.java

```diff
@@ -102,7 +108,16 @@ public void onReceive(final Object message) throws Exception {
         if (!getSender().equals(getSelf())) {
             // Public messages
             if (message instanceof ClusterEvent.CurrentClusterState) {
-                _clusterState = Optional.of((ClusterEvent.CurrentClusterState) message);
+                final ClusterEvent.CurrentClusterState clusterState = (ClusterEvent.CurrentClusterState) message;
+                _clusterState = Optional.of(clusterState);
+
+                final Metrics metrics = _metricsFactory.create();
```

Comment:
He close should be in a finally or else as a try with resources for this code to be exception safe.